kinto_http/__init__.py

```diff
@@ -259,6 +267,89 @@ def delete_buckets(self, safe=True, if_match=None, if_exists=False):
         resp, _ = self.session.request('delete', endpoint, headers=headers)
         return resp['data']
 
+    # Groups
+
+    def get_groups(self, bucket=None):
+        endpoint = self.get_endpoint('groups', bucket=bucket)
+        return self._paginated(endpoint)
+
+    def create_group(self, group=None, bucket=None,
+                     data=None, permissions=None, safe=True,
+                     if_not_exists=False):
+        if if_not_exists:
+            return self._create_if_not_exists('group',
+                                              group=group,
+                                              bucket=bucket,
+                                              data=data,
+                                              permissions=permissions,
+                                              safe=safe)
+        headers = DO_NOT_OVERWRITE if safe else None
+        endpoint = self.get_endpoint('group',
+                                     bucket=bucket,
+                                     group=group)
+        try:
+            resp, _ = self.session.request('put', endpoint, data=data,
+                                           permissions=permissions,
+                                           headers=headers)
+        except KintoException as e:
+            if e.response.status_code == 403:
+                msg = ("Unauthorized. Please check that the bucket exists and "
+                       "that you have the permission to create or write on "
+                       "this group.")
+                e = KintoException(msg, e)
+            raise e
+
+        return resp
+
+    def update_group(self, data=None, group=None, bucket=None,
+                     permissions=None, method='put',
+                     safe=True, if_match=None):
+        endpoint = self.get_endpoint('group',
+                                     bucket=bucket,
+                                     group=group)
+        headers = self._get_cache_headers(safe, data, if_match)
+        resp, _ = self.session.request(method, endpoint, data=data,
+                                       permissions=permissions,
+                                       headers=headers)
+        return resp
+
+    def patch_group(self, *args, **kwargs):
+        kwargs['method'] = 'patch'
+        return self.update_group(*args, **kwargs)
+
+    def get_group(self, group=None, bucket=None):
```

Comment:
`group=None` → Because we don't have a default name for group, we should use `group,` without the None default value. (works for all group methods)